### ISSUE TITLE
XDG_CACHE_HOME and XDG_DATA_HOME not being used, only XDG_CONFIG_HOME

### ISSUE THREAD

**Issue by i30817** [libretro/RetroArch#5311]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description
Pretty much in the title. 

### Expected behavior

Expected these three vars to be read from environment and to used to separate different kinds of files per their purpose so i could chose a appropriate filesystem or place for each.

### Actual behavior

While the current behavior is simpler to code (but not without pitfalls, see my other bug #5310 about XDG_CONFIG_HOME), the reason these are separate is to allow users to act as sysadmins of their system and store things on a appropriate filesystems or partition, or to feel safe in deleting stuff. Needless to say, downloading 1+ gb of thumbnails and storing them on a config dir, or keeping every save on the same place as the core configuration files by default is not very useful for sysadmining.

### Steps to reproduce the bug
N/A
### Bisect Results
N/A
### Version/Commit
RetroArch: Frontend for libretro -- v1.6.5 -- 5c6b230e --
Compiler: GCC (5.4.0) 64-bitBuilt: Aug 15 2017

### Environment information
Linux AIVAS 4.4.0-91-generic #114-Ubuntu SMP Tue Aug 8 11:56:56 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Comment by gouchi** [comment]
Hi,

Which data will you put in  `XDG_CACHE_HOME`  and `XDG_DATA_HOME` ?

Thank you.

**Comment by i30817** [comment]
I wrote this out when i had separate partitions because ubuntu upgrading was completely fucking awful and you had to install from pen disk and delete your partition if you wanted it to work well.

So i had a tiny 'system' partition (which probably due to a mistake of me, included 'HOME' ) and a 'data' partition, so the few gb of thumbnails were enough to fill them, and moving XDG_CONFIG_HOME broke a bunch of stuff, because programmers are completely hidebound.

Basically this doesn't matter anymore to me, but FWIW, the idea was placing the thumbnails in XDG_CACHE_HOME/retroarch/thumbnails (if XDG_CACHE_HOME existed of course, otherwise, ~/.cache/retroarch/thumbnails) then moving _that_.

You can close this if you'd like, it's not that relevant anymore.

**Comment by stacyharper** [comment]
I'd like to re-open this please:

Storing things like saves, and states, and downloaded things in XDG_CONFIG_HOME is not appropriate.

Typical use case would be to use a syncing program (ex: syncthing) to synchronize your saves across multiple devices. In the same time, the retroarch.cfg should not be the same on all devices (because of multiple distros specificities by example)

**Comment by i30817** [comment]
So be it. Sorry about the late response @stacyharper.
